portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,29 +36,45 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> comments = new ArrayList<String>();
-    comments.add("Hi tino");
-    comments.add("i like pie");
-    comments.add("nice space");
 
-    String json = convertToJson(comments);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String info = (String) entity.getProperty("info");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(name, info, timestamp);
+      comments.add(comment);
+    }
+
+    Gson gson = new Gson();
 
-    // Send the JSON as the response
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(comments));
   }
 
-    /**
-      * Converts a data array instance into a JSON string using manual String concatentation.
-    */
-  private String convertToJson(ArrayList<String> array) {
-    String json = "";
-    json += "\"" + array.get(0) + "\"";
-    json += ", ";
-    json += "\"" + array.get(1) + "\"";
-    json += ", ";
-    json += array.get(2);
-    return json;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
+    String userCommentString = request.getParameter("user-comment");
+    String userNameString = request.getParameter("user-name");
+    long timestamp = System.currentTimeMillis();
```

Comment:
Doesn't matter here, but if you were gonna execute two RPC calls before retrieving the timestamp, better do it the first thing.